core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -294,13 +315,19 @@ public String getDisplayName() {
         }
 
         /**
-         * Performs on-the-fly validation on the file mask wildcard.
+         * Performs on-the-fly validation of the file mask wildcard, when the artifacts
+         * textbox or the caseSensitive checkbox are modified
          */
-        public FormValidation doCheckArtifacts(@AncestorInPath AbstractProject project, @QueryParameter String value) throws IOException {
+        public FormValidation doCheckArtifacts(@AncestorInPath AbstractProject project,
+                @QueryParameter String value,
+                @QueryParameter(value = "caseSensitive") String caseSensitive)
+                throws IOException {
             if (project == null) {
                 return FormValidation.ok();
             }
-            return FilePath.validateFileMask(project.getSomeWorkspace(),value);
+            // defensive approach to remain case sensitive in doubtful situations
+            boolean bCaseSensitive = caseSensitive == null || !"false".equals(caseSensitive);
```

Comment:
I had this approach to be very conservative regarding existing behavior, meaning I switch to NOT case sensitive only if explicitly required. So:
 - if caseSensitive is null, I want bCaseSensitive to be true (not false!)
 - if caseSensitive is "false", I want bCaseSensitive to be false
 - if caseSensitive is "true", I want bCaseSensitive to be true
 - if caseSensitive is "tutu", I want bCaseSensitive to be true (not false!)

Boolean.parseBoolean wouldn't give me that behavior.
Of course if I'm being uselessly cautious, Boolean.parseBoolean is better :-)